docs/DEVELOPER-GUIDE.md

```diff
@@ -0,0 +1,38 @@
+# Developer Guide
+This document describes how to create a pull request for contributing to Appsody along with various ways to test code changes. 
+
+## create pull requests:
+This section describes how to create a pull request.
+* create a fork for the appsody repo, ex:
+    * https://github.com/appsody/appsody
+        * click on the Fork button
+* using git, clone your fork and create a unique branch for each issue or code task
+* after committing and pushing your changes use the appsody github webpage to create a pull request
+    * the github webpage for the upstream repo should show an indicator after pushing your changes, click the "Compare & pull request" button
+![image info](pull-request.png)
+
+* if you would like your code changes to be known/tracked but are not ready for them to be reviewed/approved you can create a draft pull request using the pull-down
+![image info](draft-pull-request.png)
+
+## testing:
+There are various ways to test your own code or someone else’s pull request (or branch):
+* test your code changes against the test bucket
+    * make test
+* build the appsody binary files for the various platforms
+    * make build-darwin (mac)
+    * make build-windows (windows)
+    * make build-linux (linux)
+        * this creates an apposdy binary in the build dir of the repo
+* test appsody binary file
+    * fully qualify the appsody binary file for the command you want to test, for example:
```

Comment:
Having read the building from source instructions for Appsody I'm surprised that we don't suggest here that the user specifies the target `localbin-{target system}` as it'd save the requirement for the linking you're suggesting? See https://github.com/appsody/appsody/blob/master/build.md#running-the-locally-built-cli

Or are we expecting that developers would prefer to have an executable named `appsody-local` to another called `appsody`?

Reply:
I think the idea is to have `appsody` run the released/installed version and then `local-appsody` would run the dev/test version. I'm not sure what `localbin-{target system}` does. Maybe @kylegc can shed some light...

Reply:
I forgot about the build.md doc. There are overlaps between it and this, but I think that is okay as both docs serve a different purpose.

The `localbin` part almost solves the same problem, except as Terrence pointed out, we want the binary to have a different name so the user can have a stable `appsody` command and an experimental `appsody-local` command at the same time.